File: age_calculator.py
from datetime import datetime

def calculate_age(birth_date, fixed_date):
    birth_date = datetime.strptime(birth_date, "%Y-%m-%d")
    fixed_date = datetime.strptime(fixed_date, "%Y-%m-%d")

    # Calculate age in years
    age = fixed_date.year - birth_date.year - ((fixed_date.month, fixed_date.day) < (birth_date.month, birth_date.day))

    # Calculate months and days
    if fixed_date.month < birth_date.month or (fixed_date.month == birth_date.month and fixed_date.day < birth_date.day):
        months = (fixed_date.month + 12) - birth_date.month - 1
    else:
        months = fixed_date.month - birth_date.month

    if fixed_date.day < birth_date.day:
        fixed_date_month_adjusted = fixed_date.replace(month=fixed_date.month - 1 if fixed_date.month > 1 else 12,
                                                      year=fixed_date.year - 1 if fixed_date.month == 1 else fixed_date.year)
        days = (fixed_date - fixed_date_month_adjusted).days
    else:
        days = fixed_date.day - birth_date.day

    return age, months, days

# Input birth date
birth_date_input = input("Enter your birth date (YYYY-MM-DD): ")

# Input fixed date for calculation
fixed_date_input = input("Enter the fixed date for calculation (YYYY-MM-DD): ")

# Calculate age
age, months, days = calculate_age(birth_date_input, fixed_date_input)

# Output the calculated age
print(f"Your age is: {age} years, {months} months, and {days} days.")

File: age_calculator.cpp
#include <iostream>

using namespace std;

int main() {
    int fixedYear, fixedMonth, fixedDay;
    int birthYear, birthMonth, birthDay;

    // Input fixed date for calculation
    cout << "Enter the fixed year for calculation: ";
    cin >> fixedYear;

    cout << "Enter the fixed month for calculation (1-12): ";
    cin >> fixedMonth;

    cout << "Enter the fixed day for calculation (1-31): ";
    cin >> fixedDay;

    // Input birth details
    cout << "Enter your birth year: ";
    cin >> birthYear;

    cout << "Enter your birth month (1-12): ";
    cin >> birthMonth;

    cout << "Enter your birth day (1-31): ";
    cin >> birthDay;

    // Calculate age
    int ageYears = fixedYear - birthYear;
    int ageMonths = fixedMonth - birthMonth;
    int ageDays = fixedDay - birthDay;

    // Adjust the age if the fixed date hasn't reached the birth date this year
    if (ageMonths < 0 || (ageMonths == 0 && ageDays < 0)) {
        ageYears--;
        ageMonths = 12 + ageMonths;
    }

    // Output the calculated age
    cout << "Your age is: " << ageYears << " years, " << ageMonths << " months, and " << ageDays << " days." << endl;

    return 0;
}
